### ISSUE TITLE
Some questions about the code

### ISSUE THREAD

**Issue by Li-Ruiqi777** [nie-lang/UDIS2#39]
Hello Dr. Nie, first of all, thank you very much for your open-source code; I have learned a lot from it.

I have a few questions regarding this part of the code in `Warp/Codes/Network.py`:

```python
def build_output_model(net, input1_tensor, input2_tensor):
    ...
    M_tensor = torch.tensor([[out_width / 2.0, 0., out_width / 2.0],
                      [0., out_height / 2.0, out_height / 2.0],
                      [0., 0., 1.]])
    N_tensor = torch.tensor([[img_w / 2.0, 0., img_w / 2.0],
                      [0., img_h / 2.0, img_h / 2.0],
                      [0., 0., 1.]])
    if torch.cuda.is_available():
        M_tensor = M_tensor.cuda()
        N_tensor = N_tensor.cuda()
    N_tensor_inv = torch.inverse(N_tensor)

    I_ = torch.tensor([[1., 0., width_min],
                      [0., 1., height_min],
                      [0., 0., 1.]])#.unsqueeze(0)
    mask = torch.ones_like(input2_tensor)
    if torch.cuda.is_available():
        I_ = I_.cuda()
        mask = mask.cuda()
    I_mat = torch.matmul(torch.matmul(N_tensor_inv, I_), M_tensor).unsqueeze(0)
    homo_output = torch_homo_transform.transformer(torch.cat((input1_tensor+1, mask), 1), I_mat, (out_height.int(), out_width.int()))
    ...
```
I would like to ask the following:

- What are the meanings of the variables `M_tensor`, `N_tensor`, `I_`, and `I_mat`? In particular, for `I_mat`, my guess is that the code applies a translation and scaling transformation to the reference image?

- Regarding `torch_homo_transform.transformer()`, I tried it myself and found that the **translation** components in the homography matrix should probably be normalized before being input. However, `I_mat` here is not normalized. Could you clarify this?

**Comment by nie-lang** [comment]
Hi there, M_tensor and  N_tensor can be regarded as the normalization matrices, which help to normalize the coordination range from [0:H, 0:W] to [-1:1, -1:1]. With M_tensor and  N_tensor, we can carry out interpolation (warping) on the normalized meshgrid.

As for I_mat, I think it has been normalized here. See:
I_mat = torch.matmul(torch.matmul(N_tensor_inv, I_), M_tensor).unsqueeze(0)
